### PR TITLE
Fixes to unblock LTP (including itimer)

### PR DESCRIPTION
**Pull request by mikbras**
This change provides fixes to unblock LTP:
- an itimer implemention
- a minor fix for ftruncate (EXT2)
- minor fixes for pipes
- a minor fix for kill (avoid when no processes to kill)

### CODE REVIEW

File: tests/ltp/Makefile [hunk 1]
```diff
@@ -1,11 +1,18 @@
+.PHONY: tests
 TOP=$(abspath ../..)
 include $(TOP)/defs.mak
 
 APPDIR=appdir
 APPBUILDER=$(TOP)/scripts/appbuilder
 
-TEST_FILE=ltp_enabled.txt
```
Comment (hukoyu) [inline tests/ltp/Makefile:7]: @mikbras , @salsal97 enabled more tests in ltp_enabled.txt in PR https://github.com/deislabs/mystikos/pull/165 which is merged. Please make sure these new tests are enabled in tests/ltp/tests

File: tests/ltp/Makefile [hunk 2]
```diff
@@ -28,7 +35,7 @@ OPTS = --strace
 endif
 
 tests:
-	@ $(foreach i, $(TESTS), $(RUNTEST) $(MYST_EXEC) $(OPTS) rootfs $(i) $(OPTS) $(NL) )
```
Comment (hukoyu) [inline tests/ltp/Makefile:31]: Thanks for fixing that.

File: tests/ltp/tests [hunk 3]
```diff
@@ -0,0 +1,1222 @@
+#/ltp/testcases/kernel/syscalls/abort/abort01
```
Comment (hukoyu) [inline tests/ltp/tests:1]: @mikbras , @salsal97 enabled more tests in ltp_enabled.txt in PR #165 which is merged. Please make sure these new tests are enabled in tests/ltp/tests

**Review by hukoyu — APPROVED** [review]
@mikbras  thanks for doing that. My only feedback is making sure to enable @salsal97 's new enabled tests in PR #165. Other than that LGTM. Thanks.

File: tests/ltp/Makefile (hunk 1 above)
Comment (mikbras) [reply tests/ltp/Makefile]: Hi Huseyin. Thank you for reviewing this. I backed out all my LTP changes and will leave this to salsal97. :)

File: tests/ltp/tests (hunk 3 above)
Comment (mikbras) [reply tests/ltp/tests]: I backed out my tests/ltp change. Thank you.

File: tests/ltp/Makefile (hunk 2 above)
Comment (mikbras) [reply tests/ltp/Makefile]: Thank you for reviewing. :)

**Comment by mikbras** [comment]
Corrupted this branch; will create new PR.

File: kernel/itimer.c [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+#include <myst/cond.h>
+#include <myst/eraise.h>
+#include <myst/mutex.h>
+#include <myst/process.h>
+#include <myst/syscall.h>
+#include <myst/time.h>
+#include <myst/timeval.h>
+
+#define USEC 1000000
+#define NSEC 1000000000
```
[the remark this answers is not in the document]
Comment (mikbras) [reply kernel/itimer.c:19]: Resolved, Thanks.

File: utils/timeval.c [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    uint64_t add;
+
+    if (x)
+        *x = 0;
+
+    if (!tv || !x)
+        ERAISE(-EINVAL);
+
+    /* reject negative timeval */
+    if (tv->tv_sec < 0)
```
[the remark this answers is not in the document]
Comment (mikbras) [reply utils/timeval.c:23]: Fixed. Thanks.

File: crt/enter.c [hunk 6]
```diff
@@ -145,3 +155,41 @@ void myst_enter_crt(void* stack, void* dynv, syscall_callback_t callback)
...
+    (void)arg;
+
+    /* Enter the kernel on the itimer thread */
+    long params[6] = { 0 };
+    myst_syscall(SYS_myst_run_itimer, params);
+
+    return NULL;
+}
+
+// Create the itimer thread in user-space and then enter the kernel with the
```
[the remark this answers is not in the document]
Comment (mikbras) [reply crt/enter.c:170]: We don't support creation of kernel-space threads yet.

File: utils/timeval.c [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        ERAISE(-EINVAL);
+
+    /* normalize the timeval */
```
[the remark this answers is not in the document]
Comment (mikbras) [reply utils/timeval.c:26]: I am not sure and I'd like to leave it because it is more readable for me anyway. :)

**Pull request closed without merging**